config.py

```diff
@@ -1129,16 +1129,22 @@ def load_check(agentConfig, hostname, checkname):
 
 def generate_jmx_configs(agentConfig, hostname, checknames=None):
     """Similar logic to load_check_directory for JMX checks"""
-    from jmxfetch import JMX_CHECKS
+    from jmxfetch import JMX_CHECKS, get_jmx_checks
+
+    jmx_checks = JMX_CHECKS
+    auto_conf_jmx_checks = get_jmx_checks()
+    for check in auto_conf_jmx_checks:
+        if check not in JMX_CHECKS:
+            jmx_checks.append(check)
```

Comment:
When `get_jmx_checks()` is [called from jmxfetch.py](https://github.com/DataDog/dd-agent/pull/3150/files#diff-64a477c26f008c29252aad6c1499c099R202) it expects a dict in return for each returned check. A set of dicts is troublesome, it may be best to keep as a list and add logic to avoid duplicates?

Will push a refactor shortly.